server/services/v1/cache/runner.go

```diff
@@ -0,0 +1,290 @@
+	}
+	for _, internalKey := range internalKeys {
+		// translate the key to user key
+		userKey := runner.encoder.DecodeInternalCacheKeyNameToExternal(internalKey)
+		_, err = runner.cacheStore.Delete(ctx, encodedCacheTableName, userKey)
+		if err != nil {
+			log.Warn().Str("cacheTableName", encodedCacheTableName).Str("cacheKey", userKey).Msg("Failed to delete cache key")
+		}
+	}
+
+	_, err = tenant.DeleteCache(ctx, tx, runner.req.GetProject(), runner.req.GetName())
+	if err != nil {
+		log.Warn().
+			Str("tenant", tenant.GetNamespace().StrId()).
+			Str("project", runner.req.GetProject()).
+			Str("cache", runner.req.GetName()).
+			Msg("Failed to update project metadata entry to delete cache")
+		return nil, ctx, err
+	}
+	return &CacheResponse{
+		Status: database.DeletedStatus,
+	}, ctx, nil
+}
+
+func (runner *SetRunner) Run(ctx context.Context, tenant *metadata.Tenant) (*CacheResponse, error) {
+	encodedCacheTableName, err := getEncodedCacheTableName(ctx, tenant, runner.req.GetProject(), runner.req.GetName(), runner.encoder)
+	if err != nil {
+		return nil, err
+	}
+
+	options := &cache.SetOptions{
+		NX: runner.req.GetNx(),
+		XX: runner.req.GetXx(),
+		EX: time.Duration(runner.req.GetEx()) * time.Second,
+	}
+	if err = runner.cacheStore.Set(ctx, encodedCacheTableName, runner.req.GetKey(), internal.NewCacheData(runner.req.GetValue()), options); err != nil {
+		return nil, errors.Internal("Failed to invoke set, reason %s", err.Error())
+	}
+	return &CacheResponse{
+		Status: SetStatus,
+	}, nil
+}
+
+func (runner *GetRunner) Run(ctx context.Context, tenant *metadata.Tenant) (*CacheResponse, error) {
+	encodedCacheTableName, err := getEncodedCacheTableName(ctx, tenant, runner.req.GetProject(), runner.req.GetName(), runner.encoder)
+	if err != nil {
+		return nil, err
+	}
+
+	options := &cache.GetOptions{
+		Expiry:    0,
+		GetDelete: false,
+	}
+	data, err := runner.cacheStore.Get(ctx, encodedCacheTableName, runner.req.GetKey(), options)
+	if err != nil {
+		if err == cache.ErrKeyNotFound {
+			return nil, errors.NotFound(err.Error())
+		}
+		return nil, err
+	}
+	return &CacheResponse{
+		Data: data.GetRawData(),
+	}, nil
+}
+
+func (runner *DelRunner) Run(ctx context.Context, tenant *metadata.Tenant) (*CacheResponse, error) {
+	encodedCacheTableName, err := getEncodedCacheTableName(ctx, tenant, runner.req.GetProject(), runner.req.GetName(), runner.encoder)
+	if err != nil {
+		return nil, err
+	}
+
+	deletedCount, err := runner.cacheStore.Delete(ctx, encodedCacheTableName, runner.req.GetKey())
+	if err != nil {
+		return nil, errors.Internal("Failed to invoke del, reason %s", err.Error())
+	}
+	return &CacheResponse{
+		Status:       DeletedStatus,
+		DeletedCount: deletedCount,
+	}, nil
+}
+
+func (runner *KeysRunner) Run(ctx context.Context, tenant *metadata.Tenant) (*CacheResponse, error) {
+	encodedCacheTableName, err := getEncodedCacheTableName(ctx, tenant, runner.req.GetProject(), runner.req.GetName(), runner.encoder)
```

Comment:
nit: It's probably obvious what getEncodedCacheTableName returns and lifetime is ~10-15 lines, so returned variables name can be shortened.